bika/lims/browser/clientfolder.py

```diff
@@ -96,93 +107,55 @@ def __init__(self, context, request):
-        # This is used to decide how much of the objects need to be waked up
-        # for further permission checks, which might get expensive on sites
-        # with many clients
-        list_pagesize = self.request.get("list_pagesize", self.pagesize)
-
-        states = {
-            'default': ['active', ],
-            'active': ['active', ],
-            'inactive': ['inactive', ],
-            'all': ['active', 'inactive']
-        }
 
-        # Use the catalog to speed things up and also limit the results
-        catalog = api.get_tool("portal_catalog")
-        catalog_query = {
-            "portal_type": "Client",
-            "inactive_state": states[state],
-            "sort_on": "sortable_title",
-            "sort_order": "ascending",
-        }
-        # Inject the searchTerm to narrow the results further
-        if searchTerm:
-            catalog_query["SearchableText"] = searchTerm
-        logger.debug("getClientList::catalog_query=%s" % catalog_query)
-        brains = catalog(catalog_query)
-
-        clients = []
-        for brain in brains:
-            # only wake up objects if they are shown on one page
-            if len(clients) > list_pagesize:
-                # otherwise append only the brain
-                clients.append(brain)
-                continue
-
-            # wake up the object
-            client = brain.getObject()
-            # skip clients where the search term does not match
-            if searchTerm and not client_match(client, searchTerm):
-                continue
-            # Only show clients to which we have Manage AR rights.
-            # (ritamo only sees Happy Hills).
-            if not mtool.checkPermission(ManageAnalysisRequests, client):
-                continue
-            clients.append(brain)
-
-        return clients
-
-    def folderitems(self):
-        self.contentsMethod = self.getClientList
-        items = BikaListingView.folderitems(self)
         registry = getUtility(IRegistry)
-        if 'bika.lims.client.default_landing_page' in registry:
-            landing_page = registry['bika.lims.client.default_landing_page']
-        else:
-            landing_page = 'analysisrequests'
-
-        for item in items:
-            if "obj" not in item:
-                continue
-            obj = item['obj']
+        self.landing_page = registry.get(self._LANDING_PAGE_REGISTRY_KEY,
+                                         'analysisrequests').encode('ascii')
 
-            item['replace']['title'] = "<a href='%s/%s'>%s</a>" % \
-                (item['url'], landing_page.encode('ascii'), item['title'])
-
-            item['EmailAddress'] = obj.getEmailAddress()
-            item['replace']['EmailAddress'] = "<a href='%s'>%s</a>" % \
-                ('mailto:%s' % obj.getEmailAddress(), obj.getEmailAddress())
-            item['Phone'] = obj.getPhone()
-            item['Fax'] = obj.getFax()
-            item['ClientID'] = obj.getClientID()
```

Comment:
Is the `ClientID` now still available as a column?

Reply:
Yes, `ClientID` is still available as a column, cause is set as such in the columns definition `self.columns` (see https://github.com/senaite/bika.lims/pull/420/files#diff-66f10ee1e8797b70edad586c9c1fb51fR62). Nevertheless, there is no need to reassign the value again here because `bika_listing` already takes care of this:

https://github.com/senaite/bika.lims/blob/a18e3fd713e837746d96cb690de68126f833fd86/bika/lims/browser/bika_listing.py#L1091-L1105